src/gt4py/stencil_object.py

```diff
@@ -178,12 +178,14 @@ def _validate_args(self, used_field_args, used_param_args, domain, origin):
                     raise ValueError(
                         f"An incompatible view was passed for field {name} to the stencil. "
                     )
-                for name_other, field_other in used_field_args.items():
-                    if field_other.mask == field.mask:
-                        if not field_other.shape == field.shape:
-                            raise ValueError(
-                                f"The fields {name} and {name_other} have the same mask but different shapes."
-                            )
+                if self.backend.startswith("dawn:"):
```

Comment:
Since this is a backend-specific test, it should be implemented in the module generated by the backend, not here. I suggest to add the check to the template using the `pre-run` variable:

https://github.com/GridTools/gt4py/blob/340a9c3ae44ff2b155f9f3898c5b8032ff162443/src/gt4py/backend/dawn_backends.py#L386